app/assets/javascripts/mumuki_laboratory/application/profile.js

```diff
@@ -26,11 +26,17 @@ mumuki.load(function() {
   });
 
   function toggleEditButtonIfThereAreChanges() {
-    if (dataChanged() || avatarChanged()) {
-      $editButton.prop('disabled', false);
-    } else {
-      $editButton.prop('disabled', true);
-    }
+    let shouldEnable = requiredFieldsAreFilled() && (dataChanged() || avatarChanged());
+
+    $editButton.prop('disabled', !shouldEnable);
+  }
+
+  function requiredFieldsAreFilled() {
+    return Object.values($userForm.find('select, textarea, input').map(function () {
+      if ($(this).prop('required')) {
+        return !!$(this).val();
+      }
+    })).every((k) => k);
```

Comment:
Here you could do something like 
```
$userForm.find('select, textarea, input').toArray().every(elem => {
      $elem = $(elem);
      return !($elem.prop('required')) || !!$elem.val(); 
})
```
to avoid calling every and map

Reply:
`!($elem.prop('required')) || !!$elem.val()`is great. :1st_place_medal: